c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -264,7 +272,12 @@ struct HirTyVisitor<'a, 'tcx> {
 }
 
 impl<'a, 'tcx> HirTyVisitor<'a, 'tcx> {
-    fn handle_ty(&mut self, rw_lty: RwLTy<'tcx>, hir_ty: &hir::Ty<'tcx>) {
+    fn handle_ty(
+        &mut self,
+        rw_lty: RwLTy<'tcx>,
+        hir_ty: &hir::Ty<'tcx>,
+        lifetime_type: LifetimeName,
+    ) {
```

Comment:
Would it make sense to include the `LifetimeName` in the `RwLTy` label instead of passing it separately?  That might make it easier to work with when there are nested pointers with different lifetimes at each level.

Reply:
I did something like this in #915 [here](https://github.com/immunant/c2rust/pull/915/files#diff-1f3bd06b6665a1221155482f5a88af065b3e807fc3fa48e543e39050416eedb3R385) (need to expand `ty.rs` to see)